src/config.rs

```diff
@@ -0,0 +1,242 @@
+    fn default() -> Self {
+        Self {
+            network: Network::Testnet,
+            public_keys: vec![],
+            anchoring_interval: 5_000,
+            transaction_fee: 10,
+            funding_transaction: None,
+        }
+    }
+}
+
+impl GlobalConfig {
+    /// Creates global configuration instance with default parameters for the
+    /// given bitcoin network and public keys of partipicants.
+    pub fn new(
+        network: Network,
+        keys: impl IntoIterator<Item = PublicKey>,
+    ) -> Result<GlobalConfig, RedeemScriptError> {
+        let public_keys = keys.into_iter().collect::<Vec<_>>();
+        if public_keys.is_empty() {
+            Err(RedeemScriptError::NotEnoughPublicKeys)?;
+        }
+
+        Ok(GlobalConfig {
+            network,
+            public_keys,
+            ..Default::default()
+        })
+    }
+
+    /// Returns the corresponding bitcoin address.
+    pub fn anchoring_address(&self) -> Address {
+        p2wsh::address(&self.redeem_script(), self.network).into()
+    }
+
+    /// Returns the corresponding redeem script.
+    pub fn redeem_script(&self) -> RedeemScript {
+        let quorum = byzantine_quorum(self.public_keys.len());
+        RedeemScriptBuilder::with_public_keys(self.public_keys.iter().map(|x| x.0))
+            .quorum(quorum)
+            .to_script()
+            .unwrap()
+    }
+
+    /// Returns the latest height below the given height which must be anchored.
+    pub fn previous_anchoring_height(&self, current_height: Height) -> Height {
+        Height(current_height.0 - current_height.0 % self.anchoring_interval)
+    }
+
+    /// Returns the nearest height upper the given height which should be anchored.
+    pub fn following_anchoring_height(&self, current_height: Height) -> Height {
+        Height(self.previous_anchoring_height(current_height).0 + self.anchoring_interval)
+    }
+}
+
+/// Local part of anchoring service configuration stored on a local machine.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct LocalConfig {
+    /// Rpc configuration. Must exist if node is validator.
+    /// Otherwise node can only check `lect` payload without any checks with `bitcoind`.
+    pub rpc: Option<BitcoinRpcConfig>,
+    /// Set of private keys for each anchoring address.
+    #[serde(with = "flatten_keypairs")]
+    pub private_keys: HashMap<Address, Privkey>,
+}
+
+/// BTC anchoring configuration.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Config {
+    /// Public part of the configuration stored in the blockchain.
+    pub global: GlobalConfig,
+    /// Local part of the configuration stored at the local machine.
+    pub local: LocalConfig,
+}
+
+mod flatten_keypairs {
+    use btc::{Address, Privkey};
+
+    use std::collections::HashMap;
+
+    /// The structure for storing the anchoring address and private key. The structure is needed to
+    /// convert data from the toml-file into memory.
```

Comment:
"...the anchoring address and the private key. The structure is required to convert data from the .toml file into memory."
Meanwhile, I am not sure "convert" is the correct word here. I guess "relocate" is closer to the meaning